cranelift/simplejit/src/backend.rs

```diff
@@ -361,8 +361,11 @@ impl<'simple_jit_backend> Backend for SimpleJITBackend {
             ref data_decls,
             ref function_relocs,
             ref data_relocs,
+            section: ref datasection,
         } = data.description();
 
+        assert!(datasection.is_none(), "Custom sections not supported");
```

Comment:
I think for the simplejit backend we'd be better off just ignoring the custom segment/section than throwing an error here.